### PR TITLE
add const to lookup

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by ibuclaw** [comment]
Looks reasonable.

**Comment by ibuclaw** [comment]
Auto-merge toggled on

File: src/dsymbol.h [hunk 1]
```diff
@@ -367,14 +367,14 @@ class DsymbolTable : public RootObject
     DsymbolTable();
 
     // Look up Identifier. Return Dsymbol if found, NULL if not.
-    Dsymbol *lookup(Identifier *ident);
+    Dsymbol *lookup(Identifier const * const ident);
```
Comment (yebblies) [inline src/dsymbol.h]: Is that what we mangle `const(Identifier)` as?

Comment (ibuclaw) [inline src/dsymbol.h]: Yes, though I think of it as what is semantically equivalent.

**Comment by JackStouffer** [comment]
Silly question: do you think there will be any noticeable increase in compile speed when all of additions of `const` are included?

**Comment by yebblies** [comment]
I doubt it.

**Pull request merged**